File: 백준/Silver/2193. 이친수/이친수.kt
fun main() {
    val n = readln().toInt()
    var answer = 0L
    var zero = 1L
    var one = 0L


    if (n > 2) {
        repeat(n - 2) {
            val newZero: Long = zero + one
            val newOne: Long = zero
    
            zero = newZero
            one = newOne
        }
    }
    
    answer = zero + one

    println(answer)
}

File: 백준/Silver/2193. 이친수/이친수.cc
#include <bits/stdc++.h>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int n;
    unsigned long long answer = 0;
    unsigned long long zero = 1;
    unsigned long long one = 0;
    cin >> n;

    if (n == 1 || n == 2) answer = 1;
    else {
        for (int i = 0; i < n - 2; i++) {
            unsigned long long newZero = zero + one;
            unsigned long long newOne = zero;

            zero = newZero;
            one = newOne;
        }
    }

    answer = zero + one;

    cout << answer << '\n';
}